WIP: Run the wagon tests directly in the core

carlobeltrame opened the pull request:
Not triggering and awaiting them in the remote repository means fewer job runners just idling to wait for some other job to complete.

To be solved is the detection of the wagon dependency repository. Also, this change is thus far untested.

carlobeltrame committed:
WIP: Run the wagon tests directly in the core

Not triggering and awaiting them in the remote repository means fewer
job runners just idling to wait for some other job to complete.

To be solved is the detection of the wagon dependency repository.
Also, this change is thus far untested.

kronn commented on .github/workflows/wagon-tests.yml:
```diff
@@ -41,6 +41,9 @@ defaults:
 
 jobs:
+  detect_wagon_dependency:
+    # TODO
```
Vorschlag:
- checkout/cd wagon-repository
- `grep "add_dependency.*hitobito_" *.gemspec | sed 's/.*hitobito_\([-_a-z]*\).*/\1/'`

Das gibt z.B. für die Jubla "youth" zurück. man kann das sed sicherlich noch anpassen, dass man direkt "hitobito_youth" zurückbekommt.

carlobeltrame replied:
Sounds good, I was thinking along the same lines. Maybe the git checkout could be sparse (only checkout the gemspec or lib dir) and only checkout the master branch.

carlobeltrame commented on .github/workflows/run-all-wagon-tests.yml:
```diff
@@ -59,39 +59,11 @@ jobs:
-      - name: Wait for ${{ matrix.wagon.name }} wagon tests result
-        uses: carlobeltrame/await-remote-run@conditional-api-requests
-        with:
-          token: ${{ secrets.WAGON_TESTS_GITHUB_TOKEN }}
```
With this change, we can also just use the default github token on line 50, no need anymore to use a PAT with write access

carlobeltrame commented:
Statuscope integration needs to be rethought with this approach, assuming each wagon has a dedicated statuscope URL to post to, which cannot easily be deduced from just the wagon name, right?

kronn commented:
for changes in the core which then fail the wagon, we could trigger a statuscope-workflow in the wagon. This would mean separating it into a workflow that solely exists to be triggered in the context of the wagon-repo.

This would result in secrets being read from the corresponding repo. The secrets then contains the token and identifier for statusscope.

The alternative would be to let said tokens live in two places. Once in the wagon for those working on features there. And once, with a slightly different naming-scheme in the core-settings to allow for fine-grained information about failures.

A third option would be to have one "wagon-tests"-identifier in statuscope that simply informs us THAT something is broken. This is better than nothing, but clearly worse than fine-grained information. OTOH, it would nicely inform WHERE the failure-info can be found.

kronn committed:
Allow workflow to determine the dependency-wagon directly

This makes the parameter obsolete. To avoid breakage, it is still provided as
'external API' of sorts, but it should really be removed.

kronn committed:
Cleanup after rebase/idea-merge
